diff_cover/violationsreporters/java_violations_reporter.py

```diff
@@ -118,6 +119,126 @@ def measured_lines(self, src_path):
         return self._info_cache[src_path][1]
 
 
+class JacocoXmlCoverageReporter(BaseViolationReporter):
+    """
+    Query information from a Jacoco XML coverage report.
+    """
+
+    def __init__(self, xml_roots, src_roots=None):
+        """
+        Load the Jacoco XML coverage report represented
+        by the cElementTree with root element `xml_root`.
+        """
+        super(JacocoXmlCoverageReporter, self).__init__("XML")
+        self._xml_roots = xml_roots
+
+        # Create a dict to cache violations dict results
+        # Keys are source file paths, values are output of `violations()`
+        self._info_cache = defaultdict(list)
+
+        if src_roots:
+            self._src_roots = src_roots
+        else:
+            self._src_roots = ['']
+
+    def _get_src_path_line_nodes(self, xml_document, src_path):
+        """
+        Return a list of nodes containing line information for `src_path`
+        in `xml_document`.
+
+        If file is not present in `xml_document`, return None
+        """
+
+        files = []
+        packages = [pkg for pkg in xml_document.findall(".//package")]
+        for pkg in packages:
+            # iterate through src_roots
+            for root in self._src_roots:
+                _files = [_file
+                          for _file in pkg.findall('sourcefile')
+                          if GitPathTool.relative_path(os.path.join(root, pkg.get('name'), _file.get('name')))
```

Comment:
Nitpick, but perhaps making this a named method can make it clearer what this is doing.
maybe something like `_get_measured_source_path(src_root, package_name, file_name)`